lightning/src/ln/channelmanager.rs

```diff
@@ -557,6 +557,9 @@ pub struct ChannelManager<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref,
 	/// See `PendingOutboundPayment` documentation for more info.
 	///
 	/// Locked *after* channel_state.
+	#[cfg(test)] // Exposed so that we can check there aren't dangling entries
+	pub(crate) pending_outbound_payments: Mutex<HashMap<PaymentId, PendingOutboundPayment>>,
+	#[cfg(not(test))]
 	pending_outbound_payments: Mutex<HashMap<PaymentId, PendingOutboundPayment>>,
```

Comment:
If we're just checking if this is empty, could we make a `#[cfg(test)]` function `has_ pending_outbound_payments` instead?